invenio_notifications/backends/utils/loaders.py

```diff
@@ -9,32 +9,41 @@
 """Template loaders for notification backend."""
 
 from flask import current_app
-from jinja2 import TemplateNotFound, TemplatesNotFound
 
 
 class JinjaTemplateLoaderMixin:
     """Used only in NotificationBackend classes."""
 
     template_folder = "invenio_notifications"
 
-    def _load_template(self, path):
-        try:
-            template = current_app.jinja_env.get_template(path)
-        except TemplateNotFound:
-            template = None
-
-        return template
-
-    def get_template(self, name):
-        """Retrieves a template from the template folder."""
-        # e.g /notifications/email/comment_edit.html
-        base_template_path = f"{self.template_folder}/{name}"
-        specific_template_path = f"{self.template_folder}/{self.id}/{name}"
-
-        base_template = self._load_template(base_template_path)
-        specific_template = self._load_template(specific_template_path)
-
-        if not specific_template and not base_template:
-            raise TemplatesNotFound(names=[base_template_path, specific_template_path])
-
-        return specific_template or base_template
+    def render_template(self, notification, recipient):
+        """Render template for a notification.
+
+        Fetch the template based on the notification type and return the template blocks.
+        More specific templates take precedence over less specific ones.
+        Rendered template will also take the locale into account.
+        """
+        # Take locale into account
+        locale = recipient.data.get("locale", "en")
+        template = current_app.jinja_env.select_template(
+            [
+                # Backend-specific templates first, e.g notifications/email/comment_edit.jinja
+                f"{self.template_folder}/{self.id}/{notification.type}.{locale}.jinja",
```

Comment:
minor: we use .html file type for the templates, but we can also leave it like this, easier for the IDEs config ;)